src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,44 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  width: $calendar-width;
+  padding: $day-padding;
+  gap: $day-gap;
+
+  &__day {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    width: $day-size;
+    height: $day-size;
+    background-color: $day-color;
+    border: 1px solid black;
+    transition: 0.5s linear;
+
+    &:hover {
+      background-color: pink;
+      transform: translate(0, -20px);
+      cursor: pointer;
+    }
+
+    @for $index from 1 through 31 {
+      &:nth-child(#{$index})::before {
+        content: "#{$index}";
+      }
+    }
+  }
+
+  @each $day, $index in $days {
+    &--start-day_#{$day} :first-child {
+      margin-left: $index * ($day-size + $day-gap);
+    }
+  }
+
+  @for $index from 28 through 31 {
+    &--month-length_#{$index} {
+      .calendar__day:first-child(n + #{$index + 1}) {
+        display: none;
+      }
```

Comment:
this loop doesn't work
<img width="1232" alt="Screenshot 2022-12-19 at 12 16 07" src="https://user-images.githubusercontent.com/94292654/208402829-1987c33a-f03f-4f8d-ad0a-2602f3c558a5.png">